plugin/environment.go

```diff
@@ -21,11 +22,17 @@ type apiImplCreatorFunc func(*model.Manifest) API
 
 type activePlugin struct {
 	BundleInfo *model.BundleInfo
-	State      int
 
 	supervisor *supervisor
 }
 
+type pluginHealthStatus struct {
```

Comment:
I'm not a fan of this name, as it now houses something out of the realm of the health check. Two appropriate options I can think of are `PluginStatus` and `PluginState`. The issue is that `PluginStatus` already [exists](https://github.com/mattermost/mattermost-server/blob/master/model/plugin_status.go) in the model module, and there is a potential confusion of `PluginState` in `Config.PluginSettings.PluginStates`, used for enabling/disabling the plugins.

Reply:
I think pluginStatus makes the most sense. The naming conflict is awkward, but it does happen where there is an external struct in the modal and another internal one for storage. 
In that case we could rename in the map in the environment too.

The other possible alternative is just to combine the two structs `activePlugin` and `pluginHealthStatus`. Supervisor would just be nil for any plugins that are not active.